cast/src/main.rs

```diff
@@ -57,12 +60,36 @@ enum Commands {
 #[tokio::main]
 async fn main() -> Result<()> {
     let cli = Cli::parse();
-
-    // todo: #2052 take network from scarb config if flag not provided
-    let network_name = cli.network.unwrap_or_else(|| {
+    let mut maybe_network_name = None;
+    if let Some(network_name) = cli.network {
+        maybe_network_name = Some(network_name);
+    } else {
+        let metadata = scarb_metadata::MetadataCommand::new()
+            .inherit_stderr()
+            .current_dir(current_dir().expect("failed to read current directory"))
+            .no_deps()
+            .exec()
+            .unwrap();
+        if metadata.packages.len() != 1 {
+            anyhow::bail!("invalid number of packages obtained from scarb metadata");
+        }
+        let package = &metadata.packages[0];
+        if let Some(tool) = &package.manifest_metadata.tool {
+            if let Some(protostar_tool) = tool.get("protostar") {
+                if let Some(network) = protostar_tool.get("network") {
+                    if let Some(network_str) = network.as_str() {
+                        maybe_network_name = Some(network_str.to_string());
+                    }
+                }
+            }
+        }
```

Comment:
Maybe something like:

```
if let Some(network_str) = package.manifest_metadata.tool
    .and_then(|tool| tool.get("protostar"))
    .and_then(|protostar_tool| protostar_tool.get("network"))
    .and_then(|network| network.as_str())
{
    maybe_network_name = Some(network_str.to_string());
}
```

?

Reply:
With this you can remove mut 
```
    let maybe_network_name = if let Some(network_name) = cli.network {
        Some(network_name)
    } else {
        let metadata = scarb_metadata::MetadataCommand::new()
            .inherit_stderr()
            .current_dir(current_dir().expect("failed to read current directory"))
            .no_deps()
            .exec()
            .unwrap();
        if metadata.packages.len() != 1 {
            anyhow::bail!("invalid number of packages obtained from scarb metadata");
        }
        let package = &metadata.packages[0];
        package
            .manifest_metadata
            .tool
            .as_ref()
            .and_then(|t| t.get("protostar"))
            .and_then(|t| t.get("network"))
            .and_then(|t| t.as_str())
            .map(|t| t.to_string())
    };
```

Reply:
Sorry for a sudden change, but we can remove it completely. We have discussed with @war-in how we can improve ux for cli, and decided that we'll remove `--network` flag altogether + use different schema in scarb.toml.
Because of this, the #2113 has been redefined, and fetching things will be done under this task, to speed up this PR merge.

Reply:
I'm going to leave it as is and you can remove it in #2113. I don't see a point in removing it here - removing it is part of task #2113.

Reply:
please remove it. I'd rather re-add it from scratch than figure out what should and what should not be done

Reply:
[revert network changes](https://github.com/software-mansion/protostar/pull/2138/commits/d1be50459470f663e30651fe860ea18034360df8)